### DISCUSSION TITLE
Client Request and Response

### DISCUSSION THREAD

**Discussion by hemalbavishi** [eclipse-iceoryx/iceoryx#1178] [Q&A] [answered]
Hello,

I have a question related to Request and Response implementation considering AUTOSAR Adaptive Platform. A service can have 0..* method with different input, output parameters. For an example if I have a service which contains 10 different method, I understand that I need to attach event for each method?

Best Regards,
Hemal Bavishi

**Comment by elBoberido** [comment, marked as answer]
Hi @hemalbavishi 
the easiest solution would be to create a server for each method and attach all this server to a listener.

Another option would be to have one server and create own message structs, e.g.
```
struct MessageBase {
    uint64_t command{0};
};

struct MessageFoo : public MessageBase {
    uint64_t dataFoo;
};
```

Then you have to set the command for each Message and circumvent the `ServiceDescription` which takes care of identifying a service. This is also only possible with the untyped API.

I would recommend to use the approach with multiple server since it is the easiest way to accomplish what you want.

**Reply by hemalbavishi** [reply]
Closing this question
